demos/marshall-2022/src/convert_to_zarr.py

```diff
@@ -38,15 +38,21 @@ def convert_h5ad_to_zarr(input_path, output_path):
     for i in range(num_cells):
         adata.obsm['X_segmentations'][i, :, :] = to_diamond(adata.obsm['X_spatial'][i, 0], adata.obsm['X_spatial'][i, 1], radius)
 
+    if isinstance(adata.X, sparse.spmatrix):
+        adata.X = adata.X.tocsc()
+
+    X = adata.X
     adata = optimize_adata(
         adata,
         obs_cols=["cell_type"],
         var_cols=["feature_name"],
         obsm_keys=["X_hvg", "X_hvg_uint8", "X_umap", "X_spatial", "X_segmentations"],
         layer_keys=[],
+        ignore_X=True
     )
+    adata.X = X
```

Comment:
I wonder if we should add a new parameter to `optimize_adata` such as `preserve_X` that does not modify `X` to avoid the need for this pattern. And we could also change `ignore_X` to `remove_X` to make it more clear what it does.